### DISCUSSION TITLE
Undefined array key occurs in WriteText when sjis is specified for SetFont

### DISCUSSION THREAD

**Discussion by naente** [mpdf/mpdf#1771] [General]
Hello, I have found what I believe to be a bug, please check it out.

# Environment
PHP 8.1
mPDF 8.1.1

# Example
```php
<?php
$mpdf = new Mpdf();
$mpdf->AddPage('P');
$mpdf->SetFont('sjis', '', 12);
$mpdf->WriteText(10, 10, 'Japanese Message');
$mpdf->Output();
```

# Error Message
> ErrorException
Undefined array key "sip" 

# Error Location
Mpdf.php at line 4423

# Proposed Amendment
```diff
- if ($this->CurrentFont['sip'] || $this->CurrentFont['smp']) {
+ if ((isset($this->CurrentFont['sip']) && $this->CurrentFont['sip']) || (isset($this->CurrentFont['smp']) && $this->CurrentFont['smp'])) {
```
In fact, in other places, the branching seems to be as above.
Ex: Line 5208, 5441

Is it possible to have the bug fixed?

**Comment by finwe** [comment]
Where was the sjis font defined? Where does the data come from?

**Reply by naente** [reply]
The process that was implemented using FPDF has been replaced to use mPDF.
sjis was specified as an argument in the current source.

If sjis is specified for SetFont, it is recognized that KozMinPro fonts will be used.
Mpdf.php SetFont > AddCJKFont > AddSJISFont > AddCIDFont
